plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,372 @@
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete a Port Channel Interface with name
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    name: ansible_port_channel_interface_changed
+    state: absent
+
+- name: Delete a Port Channel Interface with UUID
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    uuid: "{{ port_channel_interface_1.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+    format_interface_descriptions,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        name=dict(type="str", aliases=["port_channel_interface", "port_channel"]),
+        uuid=dict(type="str", aliases=["port_channel_interface_uuid", "port_channel_uuid"]),
+        description=dict(type="str"),
+        node=dict(type="str"),
+        interfaces=dict(type="list", elements="str", aliases=["members"]),
+        interface_policy_group=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+                template=dict(type="str"),
+            ),
+            aliases=["policy", "interface_policy", "interface_setting"],
+        ),
+        interface_policy_group_uuid=dict(type="str", aliases=["policy_uuid", "interface_policy_uuid", "interface_setting_uuid"]),
+        interface_descriptions=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                interface_id=dict(type="str"),
+                description=dict(type="str"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    interfaces = module.params.get("interfaces")
+    if interfaces:
+        interfaces = ",".join(interfaces)
```

Comment:
could there be issues with config when interfaces is provided as empty list? this way interfaces is of type list instead of type string

Reply:
you're right, I had an issue with that and so I made the following change:
```suggestion
    interfaces = module.params.get("interfaces")
    if isinstance(interfaces, list):
        interfaces = ",".join(interfaces)
```
This way when provided an empty list, interfaces will become an empty string and when passed into the payload and pushed to mso, we get the following error:
`MSO Error 400: {\"errors\":[\"invalid interface format in port channel ansible_port_channel_interface: Interface string can't be empty\"]}`
which is the expected behavior! Thanks for the comment